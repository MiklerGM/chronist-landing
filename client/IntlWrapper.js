import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
