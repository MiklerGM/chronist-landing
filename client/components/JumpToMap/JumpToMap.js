import React from 'react';
import { FormattedMessage } from 'react-intl';
import ym from 'react-yandex-metrika';

class JumpToMap extends React.Component {
  handleClick() {
    ym('reachGoal', 'jumpToMap');
  }

  render() {
    return (
      <div className='page--segment'>
        <div className='page--content'>
          <p>
            <FormattedMessage id='home.description' />
          </p>
          <a className='decorless' href='https://maps.chronist.ru/' onClick={() => this.handleClick()}>
            <button>
              <FormattedMessage id='home.description.button' />
            </button>
          </a>
        </div>
      </div>
    );
  }
}

export default JumpToMap;
