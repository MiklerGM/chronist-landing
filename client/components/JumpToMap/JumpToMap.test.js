import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import JumpToMap from './JumpToMap';

const TestComponent = () => (
  <IntlWrapper>
    <JumpToMap />
  </IntlWrapper>
);

// it must render
it('JumpToMap render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('JumpToMap render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('JumpToMap Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
