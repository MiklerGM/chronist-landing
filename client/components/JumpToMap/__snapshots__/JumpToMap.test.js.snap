// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JumpToMap Snapshot 1`] = `
<div
  className="page--segment"
>
  <div
    className="page--content"
  >
    <p>
      <span>
        Хронист - сервис для самостоятельного изучения исторической географии. Мы используем данные открытых источников и не претендуем на абсолютную историческую точность, но очень стремимся к ней. Помимо карты мира как основного проекта мы занимаемся интерактивными курсами и визуализациями.
      </span>
    </p>
    <a
      className="decorless"
      href="https://chronmaps.com/"
      onClick={[Function]}
    >
      <button>
        <span>
          Карта мира
        </span>
      </button>
    </a>
  </div>
</div>
`;
